sound/soc/codecs/rt711-sdca.c

```diff
@@ -267,6 +289,17 @@ static int rt711_sdca_headset_detect(struct rt711_sdca_priv *rt711)
 		break;
 	}
 
+	if (ge_mode_override) {
+		if ((det_mode != ge_mode_override) && det_mode) {
+			det_mode = ge_mode_override;
+			rt711->jack_type =
+				(rt711->jack_type == SND_JACK_HEADPHONE) ? SND_JACK_HEADSET : SND_JACK_HEADPHONE;
+			rt711_sdca_ge_force_jack_type(rt711, det_mode);
```

Comment:
optimization: I think you could move this line 297 after the if, if det_mode is zero then using rt711_sdca_ge_force_jack_type(rt711, det_mode); would work just fine.